build_tools/bazel/iree_check_test.bzl

```diff
@@ -43,19 +42,16 @@ def iree_check_test(
           driver and input file are passed automatically.
       tags: additional tags to apply to the generated test. Tag
           "driver=DRIVER" and "target=TARGET" are added automatically.
-      target_cpu_features: currently unimplemented (must be empty), will
-          eventually allow specifying target CPU features.
       timeout: timeout for the generated tests.
       **kwargs: any additional attributes to pass to the underlying native_test.
     """
 
-    if target_cpu_features:
-        fail("target_cpu_features must currently be empty")
     input_type_flags = []
     if input_type:
         input_type_flags = ["--iree-input-type=%s" % input_type]
     flags = [
         "--iree-hal-target-backends=%s" % target_backend,
+        "--iree-llvmcpu-target-cpu=generic",
     ] + compiler_flags + input_type_flags
```

Comment:
Is this implicit flag needed in the Bazel helper function? It looks like you already updated several call sites to pass it explicitly. We shouldn't always set a `llvmcpu`, even for tests using other targets.

I don't care as much about the Bazel side of that though, and it looks like CMake has the behavior that I want already.

Reply:
Good catch. There was something to change here, but to match what we do in the CMake implementation, it had to be something else. The top-level rules like `iree_check_test_suite` and `iree_generated_e2e_runner_test` take the special `target_cpu_features_variants` parameter. This defaults to `["generic"]` on CMake, so it should also default to that on Bazel. So the `"generic"` stays implicit but moves one level up within the rules.